35-Search_Insert_Position.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+'''
+Given a sorted array and a target value, return the index if the target is found.
+If not, return the index where it would be if it were inserted in order.
+
+You may assume no duplicates in the array.
+
+Example 1:
+Input: [1,3,5,6], 5
+Output: 2
+
+Example 2:
+Input: [1,3,5,6], 2
+Output: 1
+
+Example 3:
+Input: [1,3,5,6], 7
+Output: 4
+
+Example 4:
+Input: [1,3,5,6], 0
+Output: 0
+'''
+
+class Solution1:
+    # Time complexity: O(n)
+    # Space complexity:  O(1)
+    def searchInsert(self, nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: int
+        """
+        for i in range(len(nums)):
+            if nums[i] >= target:
+                return i
+        return len(nums)
+
+class Solution2:
+    # Time complexity: O(n logn)
+    # Space complexity: O(n)
+    def searchInsert(self, nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: int
+        """
+        nums.append(target)
+        return sorted(nums).index(target)
+
+
+class Solution3:
+    # Binary search method
+    # Time complexity: O(log n)
+    # Space complexity: O(1)
+    def searchInsert(self, nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: int
+        """
+        l = 0
+        r = len(nums) -1
+
+        while l <= r:
```

Comment:
Is test for equality required?

Reply:
By "<=", you're testing 1. "<" smaller than and 2. "=" equality.\
I think it will work if you don't test equality. Just smaller than should work.

Reply:
I've tried with just smaller. But for exaple, when passing the case "[1,3,5,6]  2", it failed